### PR TITLE
feat(password_policy_checks): Include password policy checks

### PR DESCRIPTION
**Pull request by n4ch04**
### Context 

Migrate from prowler 2.X to prowler 3 for checks regarding password policy, which are:

- providers/aws/services/iam/iam_password_policy_lowercase
- providers/aws/services/iam/iam_password_policy_minimum_length_14
- providers/aws/services/iam/iam_password_policy_number
- providers/aws/services/iam/iam_password_policy_reuse_24
- providers/aws/services/iam/iam_password_policy_symbol
- providers/aws/services/iam/iam_password_policy_uppercase

Since it is the same logic for all with really small changes all are included in a PR

### Description

Migrate from prowler 2.X to prowler 3 for checks regarding password policy, which are:

- providers/aws/services/iam/iam_password_policy_lowercase
- providers/aws/services/iam/iam_password_policy_minimum_length_14
- providers/aws/services/iam/iam_password_policy_number
- providers/aws/services/iam/iam_password_policy_reuse_24
- providers/aws/services/iam/iam_password_policy_symbol
- providers/aws/services/iam/iam_password_policy_uppercase

Since it is the same logic for all with really small changes all are included in a PR

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jfagoagas — CHANGES_REQUESTED** [review]
Good job @n4ch04, as an overall comments about the checks:
- We have to include the `logger` and some `try-except` clause to handle unexpected errors.
- We need to include more tests than the obvious ones for example empty responses

**Comment by n4ch04** [comment]
Since we have talk, we are not going to include the `try - except` on each check and include more detail into the exceptions catches, the same applies to empty responses

File: providers/aws/services/iam/iam_password_policy_minimum_length_14/iam_password_policy_minimum_length_14.metadata.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  "RelatedUrl": "",
+  "Remediation": {
+    "Code": {
+      "CLI": "",
+      "NativeIaC": "",
+      "Other": "",
+      "Terraform": ""
+    },
+    "Recommendation": {
+      "Text": "Ensure \"Minimum password length\" is checked under \"Password Policy\"..",
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_minimum_length_14/iam_password_policy_minimum_length_14.metadata.json]:
```suggestion
      "Text": "Ensure \"Minimum password length\" is checked under \"Password Policy\".",
```

File: providers/aws/services/iam/iam_password_policy_minimum_length_14/iam_password_policy_minimum_length_14.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+class iam_password_policy_minimum_length_14(Check):
+    def execute(self) -> Check_Report:
+        findings = []
+        report = Check_Report(self.metadata)
+        report.region = iam_client.region
+        report.resource_id = "password_policy"
+        # Check if password policy exists
+        if iam_client.password_policy:
+            # Check if symbol flag is set
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_minimum_length_14/iam_password_policy_minimum_length_14.py]:
```suggestion
            # Check password policy length
```

File: providers/aws/services/iam/iam_password_policy_number/iam_password_policy_number.metadata.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  "RelatedUrl": "",
+  "Remediation": {
+    "Code": {
+      "CLI": "",
+      "NativeIaC": "",
+      "Other": "",
+      "Terraform": ""
+    },
+    "Recommendation": {
+      "Text": "Ensure \"Require at least one number\" is checked under \"Password Policy\"..",
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_number/iam_password_policy_number.metadata.json]:
```suggestion
      "Text": "Ensure \"Require at least one number\" is checked under \"Password Policy\".",
```

File: providers/aws/services/iam/iam_password_policy_number/iam_password_policy_number.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        report = Check_Report(self.metadata)
+        report.region = iam_client.region
+        report.resource_id = "password_policy"
+        # Check if password policy exists
+        if iam_client.password_policy:
+            # Check if number flag is set
+            if iam_client.password_policy.numbers:
+                report.status = "PASS"
+                report.status_extended = (
+                    f"Password number option in password policy is set."
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_number/iam_password_policy_number.py]:
```suggestion
                    "IAM password policy require at least one number."
```

File: providers/aws/services/iam/iam_password_policy_number/iam_password_policy_number.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+                )
+            else:
+                report.status = "FAIL"
+                report.status_extended = (
+                    f"Password number option in password policy is not set."
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_number/iam_password_policy_number.py]:
```suggestion
                    "IAM password policy does not require at least one number"
```

File: providers/aws/services/iam/iam_password_policy_number/iam_password_policy_number.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+                )
+        else:
+            report.status = "FAIL"
+            report.status_extended = f"Password policy cannot be found"
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_number/iam_password_policy_number.py]:
```suggestion
            report.status_extended = "There is no password policy."
```

File: providers/aws/services/iam/iam_password_policy_minimum_length_14/iam_password_policy_minimum_length_14.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+            if (
+                iam_client.password_policy.length
+                and iam_client.password_policy.length >= 14
+            ):
+                report.status = "PASS"
+                report.status_extended = f"Password minimum length in password policy requires more than 13 characters."
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_minimum_length_14/iam_password_policy_minimum_length_14.py]:
```suggestion
                report.status_extended = "IAM password policy requires minimum length of 14 characters."
```

File: providers/aws/services/iam/iam_password_policy_minimum_length_14/iam_password_policy_minimum_length_14.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+            else:
+                report.status = "FAIL"
+                report.status_extended = f"Password minimum length in password policy requires less than 13 characters or not set."
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_minimum_length_14/iam_password_policy_minimum_length_14.py]:
```suggestion
                report.status_extended = "IAM password policy does not requires minimum length of 14 characters."
```

File: providers/aws/services/iam/iam_password_policy_lowercase/iam_password_policy_lowercase.py [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+        report = Check_Report(self.metadata)
+        report.region = iam_client.region
+        report.resource_id = "password_policy"
+        # Check if password policy exists
+        if iam_client.password_policy:
+            # Check if lowercase flag is set
+            if iam_client.password_policy.lowercase:
+                report.status = "PASS"
+                report.status_extended = (
+                    f"Password lowercase option in password policy is set."
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_lowercase/iam_password_policy_lowercase.py]:
```suggestion
                    "IAM password policy require at least one lowercase letter."
```

File: providers/aws/services/iam/iam_password_policy_lowercase/iam_password_policy_lowercase.py [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+                )
+            else:
+                report.status = "FAIL"
+                report.status_extended = (
+                    f"Password lowercase option in password policy is not set."
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_lowercase/iam_password_policy_lowercase.py]:
```suggestion
                    "IAM password policy does not require at least one lowercase letter.
```

File: providers/aws/services/iam/iam_password_policy_reuse_24/iam_password_policy_reuse_24.py [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+        # Check if password policy exists
+        if iam_client.password_policy:
+            # Check if reuse prevention flag is set
+            if (
+                iam_client.password_policy.reuse_prevention
+                and iam_client.password_policy.reuse_prevention == 24
+            ):
+                report.status = "PASS"
+                report.status_extended = (
+                    f"Password reuse prevention in password policy is equal to 24."
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_reuse_24/iam_password_policy_reuse_24.py]:
```suggestion
                    "IAM password policy reuse prevention is equal to 24."
```

File: providers/aws/services/iam/iam_password_policy_reuse_24/iam_password_policy_reuse_24.py [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+                )
+            else:
+                report.status = "FAIL"
+                report.status_extended = f"Password reuse prevention in password policy is less than 24 or not set."
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_reuse_24/iam_password_policy_reuse_24.py]:
```suggestion
                report.status_extended = "IAM password policy reuse prevention is less than 24 or not set.""
```

File: providers/aws/services/iam/iam_password_policy_reuse_24/iam_password_policy_reuse_24.py [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+        else:
+            report.status = "FAIL"
+            report.status_extended = f"Password policy cannot be found"
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_reuse_24/iam_password_policy_reuse_24.py]:
```suggestion
            report.status_extended = "There is no password policy."
```

File: providers/aws/services/iam/iam_password_policy_reuse_24/iam_password_policy_reuse_24_test.py [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+
+            check = iam_password_policy_reuse_24()
+            result = check.execute()
+            assert result[0].status == "PASS"
+
+    @mock_iam
+    def test_iam_password_policy_reuse_prevention_less_24(self):
+        iam_client = client("iam")
+        # update password policy
+        iam_client.update_account_password_policy(MinimumPasswordLength=20)
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_reuse_24/iam_password_policy_reuse_24_test.py]:
```suggestion
        iam_client.update_account_password_policy(PasswordReusePrevention=20)
```

File: providers/aws/services/iam/iam_password_policy_symbol/iam_password_policy_symbol.metadata.json [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+  "RelatedUrl": "",
+  "Remediation": {
+    "Code": {
+      "CLI": "",
+      "NativeIaC": "",
+      "Other": "",
+      "Terraform": ""
+    },
+    "Recommendation": {
+      "Text": "Ensure \"Require at least one non-alphanumeric character\" is checked under \"Password Policy\"..",
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_symbol/iam_password_policy_symbol.metadata.json]:
```suggestion
      "Text": "Ensure \"Require at least one non-alphanumeric character\" is checked under \"Password Policy\".",
```

File: providers/aws/services/iam/iam_password_policy_symbol/iam_password_policy_symbol.py [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+        report = Check_Report(self.metadata)
+        report.region = iam_client.region
+        report.resource_id = "password_policy"
+        # Check if password policy exists
+        if iam_client.password_policy:
+            # Check if symbol flag is set
+            if iam_client.password_policy.symbols:
+                report.status = "PASS"
+                report.status_extended = (
+                    f"Password symbol option in password policy is set."
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_symbol/iam_password_policy_symbol.py]:
```suggestion
                    "IAM password policy requires at least one symbol."
```

File: providers/aws/services/iam/iam_password_policy_symbol/iam_password_policy_symbol.py [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+                )
+            else:
+                report.status = "FAIL"
+                report.status_extended = (
+                    f"Password symbol option in password policy is not set."
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_symbol/iam_password_policy_symbol.py]:
```suggestion
                    "IAM password policy does not require at least one symbol."
```

File: providers/aws/services/iam/iam_password_policy_symbol/iam_password_policy_symbol.py [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+                )
+        else:
+            report.status = "FAIL"
+            report.status_extended = f"Password policy cannot be found"
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_symbol/iam_password_policy_symbol.py]:
```suggestion
            report.status_extended = "There is no password policy."
```

File: providers/aws/services/iam/iam_password_policy_uppercase/iam_password_policy_uppercase.py [hunk 19]
```diff
@@ -0,0 +1,28 @@
...
+        report = Check_Report(self.metadata)
+        report.region = iam_client.region
+        report.resource_id = "password_policy"
+        # Check if password policy exists
+        if iam_client.password_policy:
+            # Check if uppercase flag is set
+            if iam_client.password_policy.uppercase:
+                report.status = "PASS"
+                report.status_extended = (
+                    f"Password uppercase option in password policy is set."
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_uppercase/iam_password_policy_uppercase.py]:
```suggestion
                   "IAM password policy requires at least one uppercase letter."
```

File: providers/aws/services/iam/iam_password_policy_uppercase/iam_password_policy_uppercase.py [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+                )
+            else:
+                report.status = "FAIL"
+                report.status_extended = (
+                    f"Password uppercase option in password policy is not set."
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_uppercase/iam_password_policy_uppercase.py]:
```suggestion
                    "IAM password policy does not require at least one uppercase letter."
```

File: providers/aws/services/iam/iam_password_policy_uppercase/iam_password_policy_uppercase.py [hunk 21]
```diff
@@ -0,0 +1,28 @@
...
+                )
+        else:
+            report.status = "FAIL"
+            report.status_extended = f"Password policy cannot be found"
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_password_policy_uppercase/iam_password_policy_uppercase.py]:
```suggestion
            report.status_extended = "There is no password policy."
```

**Review by jfagoagas — APPROVED** [review]
Great job @n4ch04, please review the suggestions prior to merge, thanks!

File: providers/aws/services/iam/iam_password_policy_minimum_length_14/iam_password_policy_minimum_length_14.metadata.json (hunk 1 above)
Comment (n4ch04) [reply providers/aws/services/iam/iam_password_policy_minimum_length_14/iam_password_policy_minimum_length_14.metadata.json]: Done

File: providers/aws/services/iam/iam_password_policy_minimum_length_14/iam_password_policy_minimum_length_14.py (hunk 2 above)
Comment (n4ch04) [reply providers/aws/services/iam/iam_password_policy_minimum_length_14/iam_password_policy_minimum_length_14.py]: DOne

**Comment by n4ch04** [comment]
All changed and included
